<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DCWindow</class>
 <widget class="QMainWindow" name="DCWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DC reloaded</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="DCRegisterView" name="visual" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>500</width>
          <height>300</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>300</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QPlainTextEdit" name="history">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="plainText">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="command"/>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="registerGroup">
          <property name="font">
           <font>
            <family>DejaVu Sans Mono</family>
           </font>
          </property>
          <property name="title">
           <string>Register</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="labelAC">
             <property name="toolTip">
              <string>Accumulator</string>
             </property>
             <property name="text">
              <string>AC =</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="valueAC">
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="labelDR">
             <property name="toolTip">
              <string>Data Register</string>
             </property>
             <property name="text">
              <string>DR =</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="valueDR">
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="labelAR">
             <property name="toolTip">
              <string>Address Register</string>
             </property>
             <property name="text">
              <string>AR =</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="valueAR">
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="labelPC">
             <property name="toolTip">
              <string>Program Counter</string>
             </property>
             <property name="text">
              <string>PC =</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="valuePC">
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="labelSP">
             <property name="toolTip">
              <string>Stack Pointer</string>
             </property>
             <property name="text">
              <string>SP =</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="valueSP">
             <property name="text">
              <string>127</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="labelBP">
             <property name="toolTip">
              <string>Base Pointer</string>
             </property>
             <property name="text">
              <string>BP =</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="valueBP">
             <property name="text">
              <string>127</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QListView" name="RAM">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>DejaVu Sans Mono</family>
       </font>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionRun"/>
   <addaction name="actionStep"/>
   <addaction name="actionStop"/>
   <addaction name="actionAssemble"/>
   <addaction name="actionOpen"/>
  </widget>
  <action name="actionRun">
   <property name="icon">
    <iconset resource="resource-files/resources.qrc">
     <normaloff>:/icons/execute.png</normaloff>:/icons/execute.png</iconset>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>Run the current script</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="icon">
    <iconset resource="resource-files/resources.qrc">
     <normaloff>:/icons/step.png</normaloff>:/icons/step.png</iconset>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="toolTip">
    <string>One step forward</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="icon">
    <iconset resource="resource-files/resources.qrc">
     <normaloff>:/icons/stop.png</normaloff>:/icons/stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="toolTip">
    <string>Stops the execution</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Esc</string>
   </property>
  </action>
  <action name="actionAssemble">
   <property name="icon">
    <iconset resource="resource-files/resources.qrc">
     <normaloff>:/icons/assemble.png</normaloff>:/icons/assemble.png</iconset>
   </property>
   <property name="text">
    <string>Assemble</string>
   </property>
   <property name="toolTip">
    <string>Open the assembler</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="resource-files/resources.qrc">
     <normaloff>:/icons/open.png</normaloff>:/icons/open.png</iconset>
   </property>
   <property name="text">
    <string>open</string>
   </property>
   <property name="toolTip">
    <string>Open a file</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DCRegisterView</class>
   <extends>QWidget</extends>
   <header>dc.interfaces.qt.registerview</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resource-files/resources.qrc"/>
 </resources>
 <connections/>
</ui>
